packages/js-sdk/src/user/user.ts

```diff
@@ -215,6 +244,108 @@ export class User {
     return true;
   }
 
+  async _verifyAuthenticatorRetryable(
+    verify: (authenticator: MFAAuthenticator, context: VerifyContext) => Promise<string>,
+    context: VerifyContext
+  ): Promise<any> {
+    const errMsgNoCode = 'MFA code is missing.';
+
+    try {
+      const code = await verify(context.authenticator, context);
+      if (code == null) {
+        throw new KinveyError(errMsgNoCode);
+      }
+
+      const request = new KinveyHttpRequest({
+        method: HttpRequestMethod.POST,
+        auth: KinveyHttpAuth.SessionOrMaster,
+        url: formatKinveyBaasUrl(
+          KinveyBaasNamespace.User,
+          `/${this._id}/authenticators/${context.authenticator.id}/verify`
+        ),
+        body: { code },
+      });
+      const { data } = await request.execute();
+      return data;
+    } catch (err) {
+      if (err.message === errMsgNoCode || !(err instanceof KinveyError)) {
```

Comment:
Moving verify outside of the try block will remove this check and have the same behaviour.